src/engines/fx/lpc/lpc.hpp

```diff
@@ -28,17 +30,17 @@ namespace otto::engines {
 
     LPC();
 
-    audio::ProcessData<2> process(audio::ProcessData<2>) override;
+    audio::ProcessData<2> process(audio::ProcessData<1>) override;
 
   private:
-    void makeSha(gsl::span<float> buffer, float sha_period, int lag);
-    std::vector<float, MAXIMUM_LPC_ORDER> acov;
-    std::vector<float, MAXIMUM_LPC_ORDER> sigmaAndCoeffs;
-    std::vector<float, 2*MAXIMUM_LPC_ORDER> scratchBuffer;
-    std::vector<float, MAXIMUM_LPC_ORDER> prev_audio_data;
+    void makeSha(gsl::span<float> buffer, int sha_period);
+    float acov[MAXIMUM_BLOCK_SIZE];
```

Comment:
Whether or not these are supposed to be stored between process calls, they have to be allocated from `Application::current().audio_manager->buffer_pool()`. This is required for them to be alligned correctly in memory. Bonus is, you get the right size for an audio buffer.

If they are stored between process calls, keep the `AudioBufferHandle` as a member. Otherwise allocate them in each process call.

Sorry, never mind about the ones below that have size = 50, justdo those as `std::array<float, maximum_lpc_order>` instead of C-style